CommonControls/ModelImportExport/AssimpImporter.cs

```diff
@@ -0,0 +1,372 @@
+﻿using Microsoft.Xna.Framework;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using CommonControls.FileTypes.RigidModel.LodHeader;
+using CommonControls.FileTypes.RigidModel.MaterialHeaders;
+using CommonControls.FileTypes.RigidModel.Transforms;
+using CommonControls.FileTypes.RigidModel.Types;
+using CommonControls.FileTypes.RigidModel;
+using CommonControls.FileTypes.RigidModel.Vertex;
+using Assimp;
+using CommonControls.FileTypes.Animation;
+using Serilog.Formatting.Raw;
+using static CommonControls.Editors.AnimationPack.Converters.AnimationBinFileToXmlConverter;
+using CommonControls.FileTypes.Animation;
+using System.Windows.Forms;
+using System.IO;
+using Filetypes.ByteParsing;
+using static System.Windows.Forms.VisualStyles.VisualStyleElement;
+using CommonControls.FileTypes.PackFiles.Models;
+using CommonControls.Services;
+using SharpDX.Direct3D9;
+using Serilog;
+using CommonControls.Common;
+using SharpDX.MediaFoundation;
+
+namespace CommonControls.ModelImportExport
+{
+    public class AssimpImporter
+    {
+        ILogger _logger = Serilog.Log.ForContext<AssimpImporter>();
+
+        private Assimp.Scene _assScene;
+        private AnimationFile _skeletonFile;
+        private PackFileService _packFileService;
+
+        public AssimpImporter(PackFileService packFileService)
+        {
+            _packFileService = packFileService;
+        }
+        public void ImportScene(string fileName)
+        {
+            ImportAssimpScene(fileName);
+        }
+        private string GetSkeletonString()
+        {
+            string tempSkeletonString = new string("");
+
+            var parent = _assScene.RootNode;
+
+            SearchNodesRecursiveLocal(parent, ref tempSkeletonString);
+
+            return tempSkeletonString;
+
+            void SearchNodesRecursiveLocal(Node parent, ref string skeletonString)
+            {
+                foreach (var node in parent.Children)
+                {
+                    if (node.Name.Contains("skeleton"))
```

Comment:
Personal preference, but consider not using {} for one liners. It makes the reader scroll more without adding any real value with how modern IDEs work